sklearn/model_selection/_split.py

```diff
@@ -637,6 +637,121 @@ def split(self, X, y, labels=None):
         """
         return super(StratifiedKFold, self).split(X, y, labels)
 
+class HomogeneousTimeSeriesCV(_BaseKFold):
+    """Homogeneous Time Series cross-validator
+
+    Provides train/test indices to split time series data in train/test sets.
+
+    This cross-validation object is a variation of KFold.
+    In iteration k, it returns first k folds as train set and k+1 fold as
+    test set.
+
+    Read more in the :ref:`User Guide <cross_validation>`.
+
+    Parameters
+    ----------
+    n_folds : int, default=3
+        Number of folds. Must be at least 2.
+
+    Examples
+    --------
+    >>> from sklearn.model_selection import HomogeneousTimeSeriesCV
+    >>> X = np.array([[1, 2], [3, 4], [1, 2], [3, 4]])
+    >>> y = np.array([1, 2, 3, 4])
+    >>> htscv = HomogeneousTimeSeriesCV(n_folds=4)
+    >>> htscv.get_n_splits(X)
+    3
+    >>> print(htscv)  # doctest: +NORMALIZE_WHITESPACE
+    HomogeneousTimeSeriesCV(n_folds=2)
+    >>> for train_index, test_index in htscv.split(X):
+    ...    print("TRAIN:", train_index, "TEST:", test_index)
+    ...    X_train, X_test = X[train_index], X[test_index]
+    ...    y_train, y_test = y[train_index], y[test_index]
+    TRAIN: [0] TEST: [1]
+    TRAIN: [0 1] TEST: [2]
+    TRAIN: [0 1 2] TEST: [3]
+
+    Notes
+    -----
+    The first ``n_samples % n_folds`` folds have size
```

Comment:
Oh, this is related to the prev comment. Sorry I meant in "each split the number of samples are incremented by"

You can move this to the `n_folds` documentation to avoid such confusion